detectable/src/main/java/com/blackduck/integration/detectable/detectables/yarn/Version.java

```diff
@@ -15,9 +15,9 @@ public Version(int major, int minor, int patch) {
             this.patch = patch;
         }
         public Version(String[] parts) {
-            this(   parts.length > 0 && !parts[0].isBlank()? Integer.parseInt(parts[0]) : 0,
-                    parts.length > 1 && !parts[1].isBlank()? Integer.parseInt(parts[1]) : 0, 
-                    parts.length > 2 && !parts[2].isBlank()? Integer.parseInt(parts[2]) : 0);
+            this(   parts.length > 0 && parts[0] != null && !parts[0].trim().isEmpty()? Integer.parseInt(parts[0]) : 0,
```

Comment:
This is borderline asking for a `parseIntOrZero(String[] parts, int index)` method to avoid duplication, but approving since I don't feel strongly about it. :)

Reply:
Valid point. That was my first intent and remember that Java complains if there are methods in this() reference and if any lines are written before this().